cmd/configschema/configschema/resolver.go

```diff
@@ -52,7 +57,85 @@ func NewDirResolver(srcRoot string, moduleName string) DirResolver {
 }
 
 // PackageDir accepts a Type and returns its package dir.
-func (dr DirResolver) PackageDir(t reflect.Type) string {
+// If package is not found locally, searches for package location through go.mod.
+// Attempts to search through an absolute path or GOPATH
+func (dr DirResolver) PackageDir(t reflect.Type) (string, error) {
 	pkg := strings.TrimPrefix(t.PkgPath(), dr.ModuleName+"/")
+	dir := localPackageDir(dr, pkg)
+	_, err := os.ReadDir(dir)
+	if err != nil {
+		if dir, err = externalPackageDir(dr, pkg); err != nil {
+			return "", fmt.Errorf("could not find the pkg: %s", pkg)
+		}
+	}
+	return dir, nil
+}
+
+// localPackageDir returns the path to a local package.
+func localPackageDir(dr DirResolver, pkg string) string {
```

Comment:
Any reason `localPackageDir`, `externalPackageDir`, and `buildExternalPath` aren't receiver methods?

Reply:
I was thinking to only make those methods local to this package, but that would be the case as long as I did not export by capitalizing the first letter correct? Does it make more sense to turn them into resolver methods?